### PR TITLE
Added Post Routing to /potst/:id and added post filtering

### PR DESCRIPTION
**Pull request by lemohammed**
Changelog:

- fixed infinite loop useEffect bug causing excessive firebase reads
- Added selector component to HomePageView component to filter posts displayed by category
- Added Routing to PostView in PostContainer and App.tsx
- Loaded post content and displayed skeleton text in PostView accordingly
- Unsubscribed from firebase snapshot and auth state listeners after effect hook invocation

### CODE REVIEW

File: src/components/PrivateRoute.tsx [hunk 1]
```diff
@@ -18,10 +18,14 @@ export const PrivateRoute: React.FC<{
...
-        setAuthed(Boolean(user));
-        setLoading(authed == null);
-    });
+    useEffect(() => {
+        const unSubscribe = app.auth().onAuthStateChanged(function (user) {
+            setAuthed(Boolean(user));
+            setLoading(authed == null);
+        });
+        console.log('Authenticated');
+        return unSubscribe;
```
Comment (RatikKapoor) [inline src/components/PrivateRoute.tsx:27]: Does `unSubscribe` need to be invoked in a return anonymous function or will it automatically be called on return?

**Review by RatikKapoor — APPROVED** [review]
👍🏽

**Comment by lemohammed** [comment]
Both work because unsubscribe is an anonymous function itself.

File: src/components/PrivateRoute.tsx (hunk 1 above)
Comment (lemohammed) [reply src/components/PrivateRoute.tsx:27]: Both work because `unSubscribe `is an anonymous function itself.

Comment (lemohammed) [reply src/components/PrivateRoute.tsx:27]: but I'll wrap it in a function incase someone unsubs from something else in the future

**Review by RatikKapoor — DISMISSED** [review]

**Review by rjb75 — DISMISSED** [review]
Looks good!

**Review by RatikKapoor — APPROVED** [review]

**Review by rjb75 — APPROVED** [review]

**Pull request merged**